### DISCUSSION TITLE
Bresenham circle algorithm

### DISCUSSION THREAD

**Discussion by christophebraillon** [grblHAL/core#621] [General]
Hi,

I started using GRBL a few days ago for a home made CNC, previously I was developping my own code for controlling the CNC, mainly for fun and as a learning experience. I found GRBL to be much more complete (and generic), however, there are a few features in the code I developped that I think could benefit to the community. For example "exact" circle arc generation, using the Bresenham algorithm for circles [see original article](https://dl.acm.org/doi/10.1145/359423.359432).

I haven't looked deeply in the code and I would appreciate some guidance about the right way to implement this kind of feature. I think I should start with `stepper.c`. Anyway if someone is interested let me know

Christophe

**Comment by terjeio** [comment]
> I think I should start with stepper.c.

FYI _stepper.c_ (and _planner.c_) can only process linear moves. Arcs (and circles) are converted to short line segments [here](https://github.com/grblHAL/core/blob/f35d14a901db2e3aea7024d3a6d80269255dd61e/motion_control.c#L202-L395).
Passing arcs through the planner to stepper.c may not be an easy task...
